config/initializers/yabeda.rb

```diff
@@ -2,6 +2,6 @@
 
 require 'three_scale/metrics/yabeda'
 
-Rails.application.config.to_prepare do
+Rails.application.config.after_initialize do
```

Comment:
Well, actually, not sure that's the proper way :thinking: 

I was basing on the info here: https://guides.rubyonrails.org/configuring.html#using-initializer-files

Reply:
I think `after_initialize` is fine. Not sure you saw my comment though. I'm concerned about keeping state in a reloadable class.

Reply:
I've checked this. We can't wrap under `to_prepare` because that will call the `install!` method after every change. Also we can't call `ThreeScale::Metrics::Yabeda.install!` unwrapped from an initializer while it's in the autoload path (thus it's reloadable), that's deprecated.

The question here is do we want `ThreeScale::Metrics::Yabeda` to be reloadable?. If we do, then calling `after_initialize` is the right thing to do:

```
Rails.application.config.after_initialize do
  ThreeScale::Metrics::Yabeda.install!
end
```

Note there's no `require` here. In my tests, manually requiring the class makes it not-reloadable.

On the other hand, if we don't want `ThreeScale::Metrics::Yabeda` to be reloadable, then the class should be moved to `/lib` and the initializer should look like this:

```
require 'three_scale/metrics/yabeda'

ThreeScale::Metrics::Yabeda.install!
```

Reply:
Thanks @jlledom That's great info!

> Also we can't call ThreeScale::Metrics::Yabeda.install! unwrapped from an initializer while it's in the autoload path (thus it's reloadable), that's deprecated.

Do you happen to have some link to the docs? Is this the one? https://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#autoloading-during-initialization

I can't seem to reproduce this warning... But I get the idea.

Reply:
@mayorova yes exactly, it's that warning. If you leave the module in the autoload path, remove the `require`, and call `Yabeda.install!` unwrapped from an initializer, you will see the deprecation in the log files.

Reply:
yeah, that's what I did... but I checked the console, not the dev log, maybe it's that